libyul/AsmAnalysis.cpp

```diff
@@ -80,7 +80,11 @@ AsmAnalysisInfo AsmAnalyzer::analyzeStrictAssertCorrect(Dialect const& _dialect,
 vector<YulString> AsmAnalyzer::operator()(Literal const& _literal)
 {
 	expectValidType(_literal.type, _literal.location);
-	if (_literal.kind == LiteralKind::String && _literal.value.str().size() > 32)
+	if (
+		_literal.kind == LiteralKind::String &&
```

Comment:
I'm wondering if it would be a better solution to introduce a new LiteralKind for these

Reply:
Or some kind of annotation inside the AST

Reply:
Annotation sounds like a good idea. I'll try to add it. It seems more fitting to me than a new `LiteralKind` since the difference is in what they're used for not in their structure.

Reply:
Done as a special case that skips the visit of the literal. I introduced an extra method (`expectUnlimitedStringLiteral()`) which currently only contains assertions but is a good place to add checks if we ever need any.